### PR TITLE
Fixes the IV drip limits to more sane reagent_containers.

### LINKED ISSUE #25892: You can attach pills/patches to IV drips
?!?!?

### PR DESCRIPTION
**Pull request by Jalleo**
Fixes #25892
Bloodpacks /obj/item/weapon/reagent_containers/blood
Any type of drink container /obj/item/weapon/reagent_containers/food
Any type of glass container /obj/item/weapon/reagent_containers/glass (chem beakers and chem bottles)

I didnt remove the drinks and other stuff because that then maybe decided as a balance change
Edit: Ignore the above all foods are now counted because people want pizza IV drips apparently....

### CODE REVIEW

**Comment by coiax** [comment]
DOCTOR PLEASE HOOK THIS DONK POCKET UP INTO MY VIEEEENS

**Comment by Qbopper** [comment]
i can't be literally drip fed pizza -1

**Comment by Supermichael777** [comment]
was kinda silly

**Comment by Jalleo** [comment]
@Qbopper literally any food even boiled eggs......

**Comment by Qbopper** [comment]
I was kidding, this is a good PR

**Comment by Tacolizard** [comment]
This is a feature removal and I'm not being le ironic meme. Please don't remove. this is cool

**Comment by Profakos** [comment]
A single drop of sorrow shed for the time I have drained 50 units of blood into a pizza, but a drink can is a good compromise...

**Comment by PKPenguin321** [comment]
I don't like this, make it check for the OPENCONTAINER flag or bloodbags specifically, don't istype for each type of container

**Comment by duncathan** [comment]
ye whitelists like this are no bueno

**Comment by Jalleo** [comment]
Okay but I believe blood bags dont have open container flag so I am going to have to do this later today.

**Comment by RemieRichards** [comment]
@PKPenguin321 that allows (only listing the weird ones):
- Rags
- Botany spray bottles
- Salt/Pepper shakers (kind of funny)
- Plasmacutters (wat)

**Comment by Jalleo** [comment]
Right I want this from maintainers on a whole because I dont want this to be seen as a balance PR. How do I set this to be fixed?

**Comment by Cheridan** [comment]
Not a bug imo. Burger IVs are hilarious and should stay. Pills IVs make sense if they're gel capsules. Patch IVs are the only ones that are a little weird, but patches being a subtype of pill has always been bad to begin with.

**Comment by Jalleo** [comment]
@Cheridan all pills are sprited as a tablet pill.

**Comment by Jalleo** [comment]
Also when I looked at this I did originally see that basically all reagent containers need to be refactored to be much much better but that is way out of my scope and the scope of this PR.

**Comment by PKPenguin321** [comment]
Remierichards those pretty much all make sense expect maybe the rag, assuming plasma cutters have accessible liquid plasma storage

**Comment by Jalleo** [comment]
@RemieRichards want me to make a typecache of this so its more readable I literally just remembered I can do that.... (earlier today by about an hour or so)

File: code/game/machinery/iv_drip.dm [hunk 1]
```diff
@@ -7,11 +7,14 @@
 	var/mob/living/carbon/attached = null
 	var/mode = 1 // 1 is injecting, 0 is taking blood.
 	var/obj/item/weapon/reagent_containers/beaker = null
+	var/list/drip_containers = list(/obj/item/weapon/reagent_containers/blood,
```
Comment (Jalleo) [inline code/game/machinery/iv_drip.dm]: Changed to a cache setup

File: code/game/machinery/iv_drip.dm [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
+											/obj/item/weapon/reagent_containers/food/drinks,
+											/obj/item/weapon/reagent_containers/glass)
 
-
-/obj/machinery/iv_drip/New()
+/obj/machinery/iv_drip/Initialize()
```
Comment (Jalleo) [inline code/game/machinery/iv_drip.dm]: Converted to Initialize because it was requested

File: code/game/machinery/iv_drip.dm [hunk 3]
```diff
@@ -7,11 +7,14 @@
 	var/mob/living/carbon/attached = null
 	var/mode = 1 // 1 is injecting, 0 is taking blood.
 	var/obj/item/weapon/reagent_containers/beaker = null
+	var/list/drip_containers = list(/obj/item/weapon/reagent_containers/blood,
+											/obj/item/weapon/reagent_containers/food/drinks,
+											/obj/item/weapon/reagent_containers/glass)
```
Comment (Jalleo) [inline code/game/machinery/iv_drip.dm]: Just so everyone is aware the rag is a subtype of this type shall I make a blacklist specifically against the rag?
Also does anyone know of any of the drinks subtype that I should blacklist if needed?

**Comment by duncathan** [comment]
oh also this hecka lame I like IV dripping a pizza

**Comment by Jalleo** [comment]
Alright I will wait until tomorrow to see if a concensus can be made. Just work out what sort of stuff you want in it. I am fine with waiting at this bike shed.

**Comment by Jalleo** [comment]
Okay I am going to change the drinks path to be its parent instead. As this appears to be what is wanted by the maintainers. Just go to get home first.

**Pull request merged**